### ISSUE TITLE
Fix nodejs runtime

### ISSUE THREAD

**Issue by metacosm** [halkyonio/operator#191]

**Comment by lholmquist** [comment]
Anything that myself or @helio-frota might be able to help with?

**Comment by metacosm** [comment]
@lholmquist thanks but I first need to assess the situation… the problem is that halkyon expects some commands to be available in the target runtime image. For java-based runtimes, we have a specific image that we use: https://github.com/halkyonio/container-images/tree/master/hal-mvn-jdk. I need to see if we could have a better system so that we could use arbitrary runtime images, in particular the standard node.js images.

**Comment by metacosm** [comment]
@lholmquist @helio-frota: how does one access registry.access.redhat.com/ubi8/nodejs-12? Seems like halkyon cannot pull the image…

**Comment by lholmquist** [comment]
the ubi images are the "community images" and should be available to everyone

**Comment by metacosm** [comment]
Any idea why I get a image pull back off error on it then?

**Comment by lholmquist** [comment]
can you run this locally `docker run -it --rm registry.access.redhat.com/ubi8/nodejs-12 /bin/bash`

**Comment by metacosm** [comment]
The problem is that it's the operator that cannot pull the image. I can pull and execute stuff off of the image locally just fine.

**Comment by lholmquist** [comment]
> . I can pull and execute stuff off of the image locally just fine.
I just wanted to make sure it was available.   

Nodeshift actually use this `registry.access.redhat.com/ubi8/nodejs-10` which is the node 10 version. could you try that instead of 12

**Comment by lholmquist** [comment]
I've written a little node.js code that pulls this image to see if things would work from code, and it seems to work, would someone be able to verify,  https://github.com/lholmquist/docker-node-pull

**Comment by helio-frota** [comment]
That's right, @metacosm probably this is something with golang code that pulls the image and/or the external library/dependency because the code I wrote is not working https://github.com/helio-frota/try-to-pull but the library @lholmquist used works.

**Comment by metacosm** [comment]
I'll take a look tomorrow but I don't control the code that pulls the image: it's Kubernetes doing it… 😄 
There's probably a setting that we missed somewhere. I'll also take another more attentive look at the thread you mentioned because I only followed it from far away 😁

**Comment by helio-frota** [comment]
@metacosm hi, any update on this ?
No rush, but just myself to organize and sync my other tasks :+1:

**Comment by cmoulliard** [comment]
Do you plan to work on this task this week ? When will it be delivered as part of a next release ? Can you apply the label as defined here : https://github.com/snowdrop/snowdrop-team/labels?q=- ? @metacosm
